service/dynamodb/expression/operand.go

```diff
@@ -0,0 +1,251 @@
+	}
+
+	ret := ExprNode{
+		names: []string{},
+	}
+
+	nameSplit := strings.Split(p.path, ".")
+	for i, word := range nameSplit {
+		var substr string
+		if string(word[len(word)-1]) == "]" {
+			for j, char := range word {
+				if string(char) == "[" {
+					substr = word[j:]
+					word = word[:j]
+				}
+			}
+		}
+
+		ret.names = append(ret.names, word)
+		ret.fmtExpr += "$p" + substr
+		if i != len(nameSplit)-1 {
+			ret.fmtExpr += "."
+		}
+	}
+	return ret, nil
+}
+
+// BuildOperand will create the ExprNode which will be recursively
+// called in the BuildExpression operation
+func (v ValueBuilder) BuildOperand() (ExprNode, error) {
+	expr, err := dynamodbattribute.Marshal(v.value)
+	if err != nil {
+		return ExprNode{}, err
+	}
+
+	ret := ExprNode{
+		values:  []dynamodb.AttributeValue{*expr},
+		fmtExpr: "$v",
+	}
+	return ret, nil
+}
+
+// BuildOperand will create the ExprNode which will be recursively
+// called in the BuildExpression operation
+func (s SizeBuilder) BuildOperand() (ExprNode, error) {
+	ret, err := s.pb.BuildOperand()
+	ret.fmtExpr = "size (" + ret.fmtExpr + ")"
+
+	return ret, err
+}
+
+// aliasList will keep track of all the names we need to alias in the nested
+// struct of conditions and operands. This will allow each alias to be unique
+// while deduplicating aliases.
+type aliasList struct {
+	namesList  []string
+	valuesList []dynamodb.AttributeValue
+}
+
+// buildExpression returns an Expression with aliasing for paths/values specified
+// by aliasList
+func (en ExprNode) buildExprNodes(al *aliasList) (Expression, error) {
+	if al == nil {
+		return Expression{}, fmt.Errorf("buildExpression Error: aliasList is nil")
+	}
+
+	index := struct {
+		name, value, children int
+	}{}
+
+	//index := [3]int{}
+	expr := Expression{
+		Expression: en.fmtExpr,
+	}
+
+	for i := 0; i < len(expr.Expression); {
+		if expr.Expression[i] != '$' {
+			i++
+			continue
+		}
+
+		var alias string
+		switch expr.Expression[i+1] {
+		case 'p':
+			str, err := al.aliasPath(en.names[index.name])
+			if err != nil {
+				return Expression{}, err
+			}
+			alias = str
+			if expr.Names == nil {
+				expr.Names = make(map[string]*string)
+			}
+			expr.Names[alias] = &en.names[index.name]
+			index.name++
+
+		case 'v':
+			str, err := al.aliasValue(en.values[index.value])
+			if err != nil {
+				return Expression{}, err
+			}
+			alias = str
+			if expr.Values == nil {
+				expr.Values = make(map[string]*dynamodb.AttributeValue)
+			}
+			expr.Values[alias] = &en.values[index.value]
+			index.value++
+
+		case 'c':
+			childExpr, err := en.children[index.children].buildExprNodes(al)
+			if err != nil {
+				return Expression{}, err
+			}
+			alias = childExpr.Expression
+			tempExpr := expr.Expression
+			expr = mergeExpressionMaps([]Expression{expr, childExpr})
+			expr.Expression = tempExpr
+			index.children++
+		}
+		expr.Expression = expr.Expression[:i] + alias + expr.Expression[i+2:]
+		i += len(alias)
```

Comment:
we should probably have a default case that returns an error saying case not supported